ocaml/xapi/bios_strings.ml

```diff
@@ -32,26 +94,51 @@ let get_bios_string name =
     else str
   with _ -> ""
 
+let get_baseboard_strings () =
+  let keys = ["baseboard-manufacturer"; "baseboard-product-name";
+              "baseboard-version"; "baseboard-serial-number"] in
+
+  let output = get_output_of_type "2" in
+  let records = match Angstrom.parse_string P.record_list output with
+  | Ok records -> records
+  | Error msg -> []
+  in
+  let values = match records with
+  | record :: _ -> record.values
+  | [] -> warn "No BIOS records found"; []
+  in
+
+  let convert (key, value) =
+    let key = key
+      |> String.lowercase_ascii
+      |> Str.global_replace (Str.regexp " ") "-"
```

Comment:
I think we should avoid using `Str`. It is considered a bad regexp library and use `Re` instead. Here we might even use a simpler string function.

Reply:
Re's equivalent would be `|> Re.(replace_string (compile (char ' ')) ~by:"-")`

Question is, is it that much better that it needs to add a new dependency to xapi?
The benefits of using Re in this case are not apparent to me, I guess because this is not doing complex regex queries, just replacing a character in a string.

Reply:
Use `String.map`
```
val map : (char -> char) -> string -> string
String.map f s applies function f in turn to all the characters of s (in increasing index order) and stores the results in a new string that is returned.
```